### ISSUE TITLE
Forward-mode Autodiff gives wrong `nan` when argument `segment_ids[0]` is not zero  for API `tf.math.segment_max`

### ISSUE THREAD

**Issue by eeDigitalSeal** [tensorflow/tensorflow#57002]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.9

### Current Behaviour?

If the first element of `segment_ids` is not zero, the forward mode autodiff will give nan as gradient, which is incorrect. It should give the same result as reverse-mode did.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

data = tf.constant(
[[0.47682607, 0.2620497, 0.22771002],
 [0.59265005, 0.62705662, 0.81160802]], shape=(2, 3), dtype=tf.float64)


with tf.GradientTape(persistent=True) as g:
 g.watch(data)
 res_backward = tf.math.segment_max(data,[1,1])
grad_backward = g.gradient(res_backward,data)
print("reverse-mode:\n",grad_backward)

grad_fwd_arr = []

for i in range(tf.size(data)):
    tangents = tf.reshape(tf.one_hot(i,tf.size(data),dtype=tf.float64),shape=data.shape)
    with tf.autodiff.ForwardAccumulator(data, tangents) as acc:
        res_forward = tf.math.segment_max(data,[1,1])
        jvp = acc.jvp(res_forward)
        grad_fwd_arr.append(tf.reduce_sum(jvp))

grad_fwd = tf.reshape(tf.convert_to_tensor(grad_fwd_arr),shape=data.shape)
print("forward-mode:\n",grad_fwd)
```

### Relevant log output

```shell
reverse-mode:
 tf.Tensor(
[[0. 0. 0.]
 [1. 1. 1.]], shape=(2, 3), dtype=float64)
forward-mode:
 tf.Tensor(
[[nan nan nan]
 [nan nan nan]], shape=(2, 3), dtype=float64)
```
</details>

**Comment by sushreebarsa** [comment]
@eeDigitalSeal I was able to replicate the issue on colab, please find the [gist](https://colab.research.google.com/gist/sushreebarsa/22ba6326c54fb60f41144f4de3be53aa/57002.ipynb) here. Thank you!

**Comment by sachinprasadhs** [comment]
@eeDigitalSeal , Could you please refer the comment [here](https://github.com/tensorflow/tensorflow/issues/56863#issuecomment-1210173883) about gradient computation behavior in forward and backward mode.
